### ISSUE TITLE
Private messages go to the status window when the other user changes to an equivalent nick

### ISSUE THREAD

**Issue by wodim** [irssi/irssi#580]
Having a user called `Mike^`, and knowing `^` and `~` are equivalent, `/q mike~`, talk to him, and when he responds, you'll see `-!- mike~ is now known as Mike^` in the query window but his first message will appear in the status window instead of the query window.

**Comment by ailin-nemui** [comment]
is this fallout of self-message support or was it broken before (in 0.8.17 or older)

**Comment by wodim** [comment]
idk

**Comment by dequis** [comment]
Can reproduce in 0.8.15

**Comment by LemonBoy** [comment]
The fix is pretty simple, we just need to replace various `strcmp` with `server->nick_comp_func` when it is appropriate to do so.
The sloppy but quick fix is patching [this](https://github.com/irssi/irssi/blob/master/src/fe-common/core/fe-queries.c#L129) check since the casemapping-aware function covers the case where only the case differs.
The 'real' fix would be to track down the chain of events that call `query_change_nick` and make sure to use the casemapping-aware comparison when possible.

In short, fuck you `CASEMAPPING`

**Comment by ailin-nemui** [comment]
can you check if this was fixed by #586

**Comment by wodim** [comment]
It's fixed. Anyway I found a related bug: if you `/whois mike~`, you never see the "End of WHOIS" mark because Irssi does not expect the whois response for Mike^ and waits for mike~'s forever.

**Comment by ailin-nemui** [comment]
Can you please open another issue for that

**Comment by wodim** [comment]
Not really but thanks for the interest.
